models/note.go

```diff
@@ -22,16 +23,33 @@ var categoryStrings = [...]string{
 	"predictions",
 }
 
-func (category Category) String() string {
-	if category < MARGINALIA || category > PREDICTIONS {
-		return "Unknown"
+var UnDeserializeableCategoryStringError = errors.New("String does not correspond to a Note Category")
+
+func DeserializeCategory(input string) (Category, error) {
+	for i := 0; i < len(categoryStrings); i++ {
+		if input == categoryStrings[i] {
+			return Category(i), nil
+		}
 	}
+	return MARGINALIA, UnDeserializeableCategoryStringError
```

Comment:
Does go not let you return 0 here?

Reply:
IT DOESN'T AND I HATE IT!

Reply:
This took me a bit to figure out. I've been bitten by stuff like this before.

Firstly, the thing @atmiguel suggested actually does work. The following code compiles.
```go
type Category int

const (
	MARGINALIA Category = iota
	META
	QUESTIONS
	PREDICTIONS
)

var categoryStrings = []string{"A", "B", "C", "D"}

func DeserializeCategory(input string) (Category, error) {
	for i := 0; i < len(categoryStrings); i++ {
		if input == categoryStrings[i] {
			return Category(i), nil
		}
	}
	return 0, errors.New("some error")
}
```



That being said, figuring out type conversions in go can be tricky. I suspect you tried to do something innocent like this.
```go
func foo() Category {
  a := 0
  return a
}
```

^^^ that won't compile. Even though this will

```go
func foo() Category {
  return 0
}
```
This took me a bit to figure out, but I _think_ the reason the first `foo()` doesn't compile is because the compiler assigns type `int` to `a`. `int` is a **named type**, so is `Category`, and you can't assign a **named type** to another **named type**, even if they have the same **underlying type**. 

This is in contrast to
```go
type A []string

func bar() A {
  a := []string{}
  return a
}
```

^^^ that compiles fine. Here,  `A` is a named type (because I declared it with `type A`), but `a` is an **unnamed type** ([]string is unnamed). It's valid to assign an **unnamed type** to a **named type** as long as they have the same **underlying type**. In `foo`, `int` is a named type even though I didn't declare it because `int` is a **predeclared type**

Hopefully that explanation isn't too confusing. I think it'll probably be useful for you to read some of this stuff yourself
 [named type/unnamed type](https://stackoverflow.com/questions/19334542/why-can-i-type-alias-functions-and-use-them-without-casting/19334952#19334952)
[predeclared type](https://stackoverflow.com/questions/32983546/named-and-unnamed-types)
[underlying type](https://golang.org/ref/spec#Types)

also play around with some of it on your own